### PR TITLE
PP-4617 Dockerize Node.js scripts

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT

- Dockerised Node.js & build scripts so Jenkins doesn't need to be aware of Node.js / can be independent of the version

## HOW
- run `./build.sh`
- `open /build/index.html` - to view generated API docs

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,25 @@
+build-local.sh
```
Comment (rhowe-gds) [inline .dockerignore]: Sort this list?

File: build.sh [hunk 2]
```diff
@@ -37,18 +35,9 @@ else
...
-npm install
-node shins.js --minify
-
-# -- Move static files to 'build' folder --
-mkdir -p build
-cp -R index.html pub source build/
-echo -e "${green}Built pay-api-docs static files...${NC}"
+
+# -- Build docs inside docker container --
+docker build --rm -t govukpay/pay-api-docs-generator:local .
```
Comment (rhowe-gds) [inline build.sh]: This is rather racy - if we run two jobs at the same time you can't guarantee what image the following `docker run` picks up

File: docker-startup.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
```
Comment (rhowe-gds) [inline docker-startup.sh:2]: `set -eu`

File: .dockerignore (hunk 1 above)
Comment (kbottla) [reply .dockerignore]: 👍

File: docker-startup.sh (hunk 3 above)
Comment (kbottla) [reply docker-startup.sh:2]: 👍  Updated

File: build.sh (hunk 2 above)
Comment (rhowe-gds) [reply build.sh]: I don't think you're going to make this non-racy very easily so I'm OK with this as is

Comment (kbottla) [reply build.sh]: Updated to use image ID generated by docker instead. Image should now always be unique to the job

**Review by rhowe-gds — APPROVED** [review]

**Pull request merged**